hypertrace-core-graphql-gateway-service-utils/src/main/java/org/hypertrace/core/graphql/utils/gateway/UnwrappedValueConverter.java

```diff
@@ -21,6 +23,16 @@
       case DOUBLE:
         return Single.just(value.getDouble());
       case TIMESTAMP:
+        if (!StringUtils.isEmpty(value.getTimestampUnit())) {
+          if ("ns".equals(value.getTimestampUnit())) {
```

Comment:
not so clean!

Reply:
Comments about how we get the unit aside, we could rewrite this to be simpler
```java
Instant.ofEpochSecond(0, value.getTimestamp());
```

Reply:
It won't work I guess, it doesn't calculate the second from the passed nano. The nano field is meant to be an adjustment.

Reply:
It should -
```java
    public static Instant ofEpochSecond(long epochSecond, long nanoAdjustment) {
        long secs = Math.addExact(epochSecond, Math.floorDiv(nanoAdjustment, NANOS_PER_SECOND));
        int nos = (int)Math.floorMod(nanoAdjustment, NANOS_PER_SECOND);
        return create(secs, nos);
    }
```

Reply:
ah right, which implementation did I check then 😅